test/logger_test.coffee

```diff
@@ -0,0 +1,205 @@
+kv  = require "../lib/kayvee"
+logger = require "../lib/logger/logger"
+assert = require 'assert'
+_ = require 'underscore'
+_.mixin = require 'underscore.deep'
+fs = require 'fs'
+
+sample = ""
+outputFunc = (text) -> sample = text
+
+describe 'logger_test', ->
+  logObj = {}
+  logObj2 = {}
+  beforeEach ->
+  	logObj = new logger("logger-tester")
+  	sample = ""
+  	logObj.setOutput outputFunc
+
+  describe '.constructor', ->
+    it 'constructor', ->
+      formatter = (data) -> data["level"] + "." + data["source"] + "." + data["title"]
+      logObj = new logger("logger-test", logger.Info, formatter, outputFunc)
+      logObj.debug "testlogdebug"
+      expected = ""
+      assert.equal sample, expected
+
+      logObj.info "testloginfo"
+      expected = logger.Info + ".logger-test.testloginfo"
+      assert.equal sample, expected
+  describe '.validatelog', ->
+    # Explicit validation checks
+    it 'validatelog', ->
+      logLvl = logObj._validateLogLvl "debug"
+      assert.equal logLvl, logger.Debug
+      logLvl = logObj._validateLogLvl "Debug"
+      assert.equal logLvl, logger.Debug
+
+      logLvl = logObj._validateLogLvl "info"
+      assert.equal logLvl, logger.Info
+      logLvl = logObj._validateLogLvl "Info"
+      assert.equal logLvl, logger.Info
+
+      logLvl = logObj._validateLogLvl "sometest"
+      assert.equal logLvl, logger.Debug
```

Comment:
This comment applies throughout. e.g.

similarly, describing in a sentence would make things more clear.

```
describe diffoutput 
  it does something
```